### PR TITLE
Fix #7466: Round Doubles when displaying Deck Options

### LINKED ISSUE #4448: "Interval Modifier" (and other percentage based) has precision issue: type 90 -> got 89%
###### Research

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum
###### Reproduction Steps
1. Long click on any desk
2. Tap `Desk options`
3. Tap `Reviews`
4. Tap `Interval modifier`
5. Type `90`
6. Tap `OK`
###### Expected Result

Interval modifier
90%
###### Actual Result

Interval modifier
89%

The same bug in any other percentage based desk prefs, like `Desk options` -> `Lapses` -> `New interval`. 
###### Debug info

AnkiDroid Version = 2.6.1
Android Version = 4.4.2
ACRA UUID = f356631b-222e-40ff-9a97-893bc3c5baaf

### LINKED ISSUE #7466: Deck Options NumberRangePreference displays wrong value due to rounding issue
When you modify some of the deck options like Reviews/Easy Bonus or Reviews/Interval wrong value is displayed. 

For example if you modify Easy Bonus to 130 the actual value that is shown in the UI is 129%. This only happens for some values and must be related to some rounding issue (nothing major). 

I can write a PR that will fix it.

### PR DESCRIPTION
**Pull request by Volfee**
## Purpose / Description
When you modify some of the deck options like Reviews/Easy Bonus or Reviews/Interval wrong value is displayed.

For example if you modify Easy Bonus to 130 the actual value that is shown in the UI is 129%. This only happens for some values and must be related to some rounding issue.

## Fixes
Fixes #7466
Fixes #4448

## Approach
* Changing  Integer.toString() method to String.format("%.0f", ...) that rounds Double values into nearest number without decimals.

## How Has This Been Tested?
* Emulated device Pixel 3a SDK 29.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java [hunk 1]
```diff
@@ -106,6 +107,7 @@ public DeckOptions getDeckOptions() {
         }
 
 
+        @SuppressLint("DefaultLocale")
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java:110]: I just learned about this a few days ago, but I think you can use `Locale.ROOT` for locale-sensitive operations where you do not actually care about the locale? https://developer.android.com/reference/java/util/Locale#ROOT

**Review by mikehardy — CHANGES_REQUESTED** [review]
added one quick note that I think will get rid of the deprecation warning, but otherwise this looks good. Could you post a screenshot of a before/after since this affects the UI? (helps us make sure, without pulling the branch to test)

**Comment by Volfee** [comment]
Thanks Mike nice finding! I have changed the code to use Local.ROOT

Re screenshots: I think, in this case it is a bit difficult to demonstrate change with SS as there is no change to the UI itself, but rather a value in form is populated correctly now ie. 130 instead of 129 when user types 130. 

If you think SS would be useful anyways I can add them to the thread.

**Review by mikehardy — APPROVED** [review]
ok - LGTM - thanks for fixing this!

**Comment by mikehardy** [comment]
merged, will be out in the next alpha - @Volfee -> 🏆  😄

**Comment by mikehardy** [comment]
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users to support maintenance and development of AnkiDroid

This PR was included in 2.14, and if you'd like to submit an expense to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!

**Pull request merged**